poly-commitment/src/ipa.rs

```diff
@@ -408,18 +408,36 @@ where
     ) -> PolyComm<G> {
         let is_zero = plnm.is_zero();
 
-        let coeffs: Vec<_> = plnm.iter().map(|c| c.into_bigint()).collect();
-
         // chunk while committing
-        let mut chunks = vec![];
-        if is_zero {
-            chunks.push(G::zero());
+        let mut chunks: Vec<_> = if is_zero {
+            vec![G::zero()]
+        } else if plnm.len() < self.g.len() {
+            vec![G::Group::msm(&self.g[..plnm.len()], &plnm.coeffs)
+                .unwrap()
+                .into_affine()]
+        } else if plnm.len() == self.g.len() {
+            // when processing a single chunk, it's faster to parallelise vertically in 2 threads
```

Comment:
> it's faster to parallelise vertically in 2 threads

Ah? Did you check experimentally?
I would mostly say that it depends on the size.

Reply:
Yes. I need to refer to a benchmark.